### PR TITLE
proxy/ipvs: test cleanLegacyService with real servers

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
There was a recent bug fixed in https://github.com/kubernetes/kubernetes/pull/80942 where a VirtualServer is not deleted if it has real servers for up to 15m. This adds a testcase for `cleanLegacyService` where a VirtualServer has backends. 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by andrewsykim** [comment]
/assign @lbernail 

This test will fail until https://github.com/kubernetes/kubernetes/pull/80942 is merged

File: pkg/proxy/ipvs/proxier_test.go [hunk 1]
```diff
@@ -2926,6 +2926,99 @@ func TestCleanLegacyService(t *testing.T) {
...
+		"ipvs2": { // not deleted
+			Address:   net.ParseIP("3.3.3.3"),
+			Protocol:  string(v1.ProtocolUDP),
+			Port:      54,
+			Scheduler: "rr",
+			Flags:     utilipvs.FlagHashed,
+		},
+	}
+
+	// "ipvs0" has a real server so it shouldn't be deleted.
```
Comment (lbernail) [inline pkg/proxy/ipvs/proxier_test.go]: We now delete the VS and unbind the IP when we delete the service, regardless of RealServers right?
(the code below seems to do exactly this so I assume the comment is wrong?)

Comment (andrewsykim) [reply pkg/proxy/ipvs/proxier_test.go]: ah yeah, sorry copy/paste error, will fix

**Comment by lbernail** [comment]
lgtm
(I'm just waiting for the fix PR to get merged for the real one)

**Comment by andrewsykim** [comment]
had to fix gofmt

**Pull request merged**